src/Symfony/Component/Messenger/Transport/Sender/SendersLocator.php

```diff
@@ -51,6 +51,12 @@ public function getSenders(Envelope $envelope): iterable
 
         foreach (HandlersLocator::listTypes($envelope) as $type) {
             foreach ($this->sendersMap[$type] ?? [] as $senderAlias) {
+                if ('*' === $type && $seen) {
```

Comment:
I'd suggest using `'*' === $type[-1]` to account for #48531

Reply:
Code updated. I'm not sure if this also affects that PR though, but we'll find out whenever either PR is merged in 6.3 :)